### PR TITLE
update spec

### LINKED ISSUE #179: specification out of date
Forgot to update the spec in latest language change.

### PR DESCRIPTION
**Pull request by tiago4orion**
Closes #179

### CODE REVIEW

File: spec.ebnf [hunk 1]
```diff
@@ -28,20 +28,20 @@ builtin = importDecl | rforkDecl | ifDecl | forDecl | setenvDecl |
...
-ifDecl = "if" ( identifier | varValue ) comparison
-              ( identifier | varValue ) "{" program "}"
+ifDecl = "if" ( variable | string ) comparison
+              ( variable | string ) "{" program "}"
          [ "else" "{" program "}" ]
          [ "else" ifDecl ] .
 
 /* For loop */
-forDecl = "for" [ identifier "in" list ] "{" program "}" .
+forDecl = "for" [ identifier "in" ( list | variable | fnInv) ] "{" program "}" .
```
Comment (katcipis) [inline spec.ebnf]: for x in func() ?

Awesome :-)

File: spec.ebnf [hunk 2]
```diff
@@ -72,7 +72,7 @@ setenvDecl = "setenv" ( identifier | varDecl ) .
 comment = "#" { unicode_char } .
 
 /* Lists */
-list = "(" { varValue } ")" .
+list = "(" { argument } ")" .
```
Comment (katcipis) [inline spec.ebnf]: more clear :+1:

File: spec.ebnf [hunk 3]
```diff
@@ -91,11 +91,11 @@ variable    = "$" identifier .
 
 comparison  = "==" | "!=" .
 
-string_lit   = "\"" { unicode_char | newline } "\"" .
+stringLit   = "\"" { unicode_char | newline } "\"" .
 
-stringConcat = ( string_lit "+" string_lit ) |
-               ( variable "+" string_lit ) |
-               ( string_lit "+" variable ) .
+stringConcat = ( stringLit "+" stringLit ) |
```
Comment (katcipis) [inline spec.ebnf]: I'm REALLY rusty on my BNF skills, but can't this be written like:

```
stringConcat = ( stringLit | variable ) "+" (stringLit | variable )
```

?

Comment (tiago4orion) [inline spec.ebnf]: hahaha holy shit!

Thanks!!!

Comment (katcipis) [inline spec.ebnf]: BNFs are FUN :-)

The way you where doing it would get really messy with more combinations.

Comment (tiago4orion) [inline spec.ebnf]: I knew I was making shit.. but lazy to think more about kkk

Comment (katcipis) [inline spec.ebnf]: The BNF looks great ;-)

Comment (tiago4orion) [inline spec.ebnf]: forgot to change this kkk :joy: 
sorry...

I'll change in #172

**Pull request merged**